scripts/wsrep_sst_clone.sh

```diff
@@ -328,85 +320,157 @@ setup_clone_plugin()
-    [ "$SERVER_SSL_CA" = "NULL" ] && SERVER_SSL_CA=
-
-    if [ "$CLONE_SSL_CERT" == "NULL" -o "$CLONE_SSL_CERT" = "" ]
+    # Server info
+    SERVER_SSL_CERT=$($MYSQL_ACLIENT -e "SELECT @@ssl_cert")
+    SERVER_SSL_KEY=$($MYSQL_ACLIENT -e "SELECT @@ssl_key")
+    SERVER_SSL_CA=$($MYSQL_ACLIENT -e "SELECT @@ssl_ca")
+    DATA=$($MYSQL_ACLIENT -e "SELECT @@datadir")
+    PXC_ENCRYPT_CLUSTER_TRAFFIC=$($MYSQL_ACLIENT -e "SELECT @@pxc_encrypt_cluster_traffic")
+    PXC_ENCRYPT_CLUSTER_TRAFFIC=$(normalize_boolean "$PXC_ENCRYPT_CLUSTER_TRAFFIC" "on")
+    CLIENT_SSL_VALID="no"
+    # We set yes as default for the server given the joiner
+    # will not need to perform these checks and we get them for granted
+    SERVER_SSL_VALID="yes"
+
+    # If pxc_encrypt_cluster_traffic is set ON then we go through the ssl checks
+    # otherwise we skip and do not set SSL    
+    wsrep_log_debug ":->(PRE) CLIENT_SSL_VALID=$CLIENT_SSL_VALID SERVER_SSL_VALID=$SERVER_SSL_VALID PXC_ENCRYPT_CLUSTER_TRAFFIC=$PXC_ENCRYPT_CLUSTER_TRAFFIC"
+    if [ "$PXC_ENCRYPT_CLUSTER_TRAFFIC" == "on" ]
     then
-        wsrep_log_info "CLONE SSL not configured. Checking general MySQL SSL settings."
+        # Client information must check on both side to allow connection from Donor to Joiner
+        wsrep_log_debug "-> CLONE_SSL_CERT: $CLONE_SSL_CERT; \ 
+        CLONE_SSL_KEY: $CLONE_SSL_KEY; CLONE_SSL_CA: $CLONE_SSL_CA"
 
-        if [ "$ROLE" = "donor" ]
+        if [ "$CLONE_SSL_CERT" == "NULL" ] || [ "$CLONE_SSL_CERT" = "" ]
         then
-            CLONE_SSL_CERT=$SERVER_SSL_CERT
-            CLONE_SSL_KEY=$SERVER_SSL_KEY
-            CLONE_SSL_CA=$SERVER_SSL_CA
-        else
+            CLONE_SSL_CERT=$(parse_cnf mysqld clone_ssl_cert "")
+            CLONE_SSL_KEY=$(parse_cnf mysqld clone_ssl_key "")
+            CLONE_SSL_CA=$(parse_cnf mysqld clone_ssl_ca "")
+        fi
+
+        if [ "$CLONE_SSL_CERT" == "NULL" ] || [ "$CLONE_SSL_CERT" = "" ]
+        then
+            wsrep_log_info "CLONE SSL not configured. Checking general MySQL SSL settings."
             CLONE_SSL_CERT=$CLIENT_SSL_CERT
             CLONE_SSL_KEY=$CLIENT_SSL_KEY
             CLONE_SSL_CA=$CLIENT_SSL_CA
-        fi
 
-        if [ -n "$CLONE_SSL_CERT" -a -n "$CLONE_SSL_KEY" ]
-        then
-            wsrep_log_info "Using SSL configuration from MySQL Server."
-            $MYSQL_ACLIENT -e "SET GLOBAL clone_ssl_cert='$CLONE_SSL_CERT'"
-            $MYSQL_ACLIENT -e "SET GLOBAL clone_ssl_key='$CLONE_SSL_KEY'"
-            $MYSQL_ACLIENT -e "SET GLOBAL clone_ssl_ca='$CLONE_SSL_CA'"
-            if [ "$CLEANUP_CLONE_PLUGIN" == "yes" ]; then
-                CLEANUP_CLONE_SSL="yes"
+            # We actually modify the variables only if we are on the Joiner
+            if [ -n "$CLONE_SSL_CERT" ] && [ -n "$CLONE_SSL_KEY" ] && [ "$ROLE" == "joiner" ]
+            then
+                wsrep_log_info "Using SSL configuration from MySQL Server."
```

Comment:
Let's say donor has clone plugin installed and set server certificates. `pxc_encrypt_cluster_traffic=off`. This means:
1. cloning from donor will require clone client to use secure connection
2. galera communication channel is not encrypted
Am I right?

Havig the above logic on Joiner side, clone plugin will be installed, but `clone_ssl_*` will not be set.

In my opinion the logic on joiner node should be something like:
1. Check if clone plugin installed. If not -> install it
2. Check if clone_ssl_ certificates set. If set -> leave as it is
3. if not set -> use certificates from [sst] section, but if pxc_encrypt_cluster_traffic=on use certificates from [mysqld] section if available